babel/src/main/java/org/apache/calcite/sql/babel/SqlColumnAttributeDateFormat.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to you under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.calcite.sql.babel;
+
+import org.apache.calcite.sql.SqlColumnAttribute;
+import org.apache.calcite.sql.SqlNode;
+import org.apache.calcite.sql.SqlWriter;
+import org.apache.calcite.sql.parser.SqlParserPos;
+
+/**
+ * A {@code SqlColumnAttributeDateFormat} is an AST node that gives
+ * a Date column of a specific format
+ */
+public class SqlColumnAttributeDateFormat extends SqlColumnAttribute {
+
+  private SqlNode formatString = null;
+
+  /**
+   * Creates a {@code SqlCharacterSetToCharacterSet}.
+   *
+   * @param pos  Parser position, must not be null
+   * @param formatString  The format string string literal
+   */
+  public SqlColumnAttributeDateFormat(SqlParserPos pos, SqlNode formatString) {
+    super(pos);
+    this.formatString = formatString;
+  }
+
+  @Override public void unparse(final SqlWriter writer, final int leftPrec,
+      final int rightPrec) {
+    if (formatString != null) {
```

Comment:
I'm a bit confused as to how formatString can ever be null. Isn't this object only created when \<FORMAT\> \<string\> is parsed?

Reply:
The way the code is currently laid out, I agree it can never be null. But as the parsing code changes over time, there are no guarantees of that. And having guards like this in place are usually a good idea to handle against future changes.

Reply:
You are correct. The null check was a leftover piece of code from my previous trail approach. It is not needed here. Removed in https://github.com/googleinterns/calcite/pull/103/commits/759efab9843a94cc2467c3e2d47333e5b1c7ef5a.